sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/TestingServiceVersions.java

```diff
@@ -0,0 +1,83 @@
+ * {@link TestMode#RECORD}, {@link TestMode#LIVE}, and {@link TestMode#PLAYBACK} modes.
+ * <p>
+ * The service versions configured here mat interact with how {@link AzureMethodSource} generates parameterized test
+ * permutations.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+@Inherited
+public @interface TestingServiceVersions {
+    /**
+     * The class that represents the service version type.
+     * <p>
+     * This is used to convert {@link #minimumLiveServiceVersion()}, {@link #maximumLiveServiceVersion()}, and {@link
+     * #recordingServiceVersion()} into their {@link ServiceVersion} type.
+     * <p>
+     * If neither of {@link #minimumLiveServiceVersion()} and {@link #maximumLiveServiceVersion()} are provided all
+     * {@link ServiceVersion ServiceVersions} represented by this type will be used. If {@link
+     * #recordingServiceVersion()} isn't configured the latest {@link ServiceVersion} represented by this type will be
+     * used.
+     * <p>
+     * If this type isn't an {@link Enum} an {@link IllegalArgumentException} will be thrown during runtime.
+     *
+     * @return The class that represents the service version type.
+     */
+    Class<? extends ServiceVersion> serviceVersionType();
+
+    /**
+     * The minimum service version used during {@link TestMode#LIVE} testing.
+     * <p>
+     * If neither this or {@link #maximumLiveServiceVersion()} are set all service versions will be used. If only this
+     * is set then all service versions equal to or later than the set version will be used. If both this and {@link
+     * #maximumLiveServiceVersion()} are set then the inclusive range of service versions will be used.
+     * <p>
+     * If no service versions meet the requirements of this and {@link #maximumLiveServiceVersion()} an {@link
+     * IllegalStateException} will be thrown during runtime.
+     * <p>
+     * {@link ServiceVersion ServiceVersions} are compared using their enum ordinal.
+     *
+     * @return The minimum service version that will be used.
+     */
+    String minimumLiveServiceVersion() default "";
+
+    /**
+     * The maximum service version used during {@link TestMode#LIVE} testing.
+     * <p>
+     * If neither this or {@link #minimumLiveServiceVersion()} are set all service versions will be used. If only this
+     * is set then all service versions less than or equal to the set version will be used. If both this and {@link
+     * #minimumLiveServiceVersion()} are set then the inclusive range of service versions will be used.
+     * <p>
+     * If no service versions meet the requirements of this and {@link #minimumLiveServiceVersion()} an {@link
+     * IllegalStateException} will be thrown during runtime.
+     * <p>
+     * {@link ServiceVersion ServiceVersions} are compared using their enum ordinal.
+     *
+     * @return The minimum service version that will be used.
+     */
+    String maximumLiveServiceVersion() default "";
```

Comment:
If we are using ordinals, should this return an `int` then?

Reply:
It could either be an `int` or `String` depending on what we want to base this on. `int` would be an ordinal base and `String` would be the enum name based. I think given where this is being configured using the enum name makes the test clearer on what it is doing.

Reply:
yeah, agree, if we are using the enum name, then String makes sense and also makes the test more readable.